pkg/plugin/system/system_test.go

```diff
@@ -276,6 +280,307 @@ func TestPluginSystemInterceptors(t *testing.T) {
 		})
 
 		g.Expect(err).NotTo(BeNil())
+		g.Expect(err.(*aspecterrors.ExitError).Err).To(MatchError("interceptor error"))
 		g.Expect(err.(*aspecterrors.ExitError).ExitCode).To(Equal(1))
 	})
 }
+
+func TestConfigure(t *testing.T) {
+	t.Run("fails when Finder fails to find plugin config file", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		finder := loader_mock.NewMockFinder(ctrl)
+		finder.EXPECT().Find().Return("", errors.New("finder"))
+
+		ps := &pluginSystem{finder: finder}
+
+		err := ps.Configure(streams)
+
+		g.Expect(err).To(MatchError("failed to configure plugin system: finder"))
+	})
+
+	t.Run("fails when Parser fails to parse plugin config file", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		finder := loader_mock.NewMockFinder(ctrl)
+		finder.EXPECT().Find().Return("/foo/bar", nil)
+
+		parser := loader_mock.NewMockParser(ctrl)
+		parser.EXPECT().Parse("/foo/bar").Return(
+			[]loader.AspectPlugin{},
+			errors.New("parser"),
+		)
+
+		ps := &pluginSystem{
+			finder: finder,
+			parser: parser,
+		}
+
+		err := ps.Configure(streams)
+
+		g.Expect(err).To(MatchError("failed to configure plugin system: parser"))
+	})
+
+	t.Run("works when 0 plugins are found in config file", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		finder := loader_mock.NewMockFinder(ctrl)
+		finder.EXPECT().Find().Return("/foo/bar", nil)
+
+		parser := loader_mock.NewMockParser(ctrl)
+		parser.EXPECT().Parse("/foo/bar").Return([]loader.AspectPlugin{}, nil)
+
+		ps := &pluginSystem{
+			finder: finder,
+			parser: parser,
+		}
+
+		err := ps.Configure(streams)
+
+		g.Expect(err).To(BeNil())
+	})
+
+	t.Run("creates and persists each plugin after invoking plugin.Setup()", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		testPlugin := loader.AspectPlugin{
+			Name:     "test plugin",
+			From:     "...",
+			LogLevel: "debug",
+		}
+		testPlugin2 := loader.AspectPlugin{
+			Name:     "test plugin2",
+			From:     "...",
+			LogLevel: "debug",
+		}
+
+		p1 := plugin_mock.NewMockPlugin(ctrl)
+		p1.EXPECT().Setup(gomock.Any())
```

Comment:
In a follow-up PR, it would be nice to be explicit about what the Setup takes. I.e. testing that it's getting called with the expected args.

Reply:
I've updated some of them, but this specific one accepts an array of bytes which is pretty awkward. I think I'd only want to verify the bytes are correct in a test explicitly testing that, to avoid the "data => bytes" logic being repeated everywhere.